core/src/main/scala/aima/core/agent/Actuator.scala

```diff
@@ -5,15 +5,14 @@ import aima.core.random.Randomness
 /**
   * @author Shawn Garner
   */
-trait Actuator[Action, Percept] {
-  def agent: Agent[Action, Percept]
-  def act(action: Action, environment: Environment[Action, Percept]): Environment[Action, Percept]
+trait Actuator[E, A] {
+  def act(action: A, e: E): E //could be Action => E => E
```

Comment:
@benderpremier I feel like making it e:E  is just too generic. I can't figure out from looking at it what it means.  I would need a decoding manual to figure out what E represents.  I like it the way it was before.  Also we lost the notion of Percept and Environment and Action.  I don't think this will help learning or matching up things to the algorithms in the book.

Reply:
@benderpremier are all these changes needed just for scala 2.13 and scala js?  Could we actually do these in smaller separate PRs?

Reply:
Hi @BusyByte,

Sorry the last two commits weren't meant for this branch. I didn't pay attention when pushing it yesterday. Let me correct that.

Reply:
Done, it should be easier to review now. Sorry about that :)